pkg/securitycontextconstraints/group/mustrunas.go

```diff
@@ -5,33 +5,54 @@ import (
 
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/securitycontext"
 
 	securityv1 "github.com/openshift/api/security/v1"
 )
 
 // mustRunAs implements the GroupSecurityContextConstraintsStrategy interface
 type mustRunAs struct {
-	ranges []securityv1.IDRange
-	field  string
+	ranges        []securityv1.IDRange
+	field         string
+	groupAccessor GroupAccessorFunc
+	groupMutator  GroupMutatorFunc
 }
 
 var _ GroupSecurityContextConstraintsStrategy = &mustRunAs{}
 
+type GroupAccessorFunc func(securitycontext.PodSecurityContextAccessor) []int64
+type GroupMutatorFunc func(securitycontext.PodSecurityContextMutator, int64)
+
 // NewMustRunAs provides a new MustRunAs strategy based on ranges.
-func NewMustRunAs(ranges []securityv1.IDRange, field string) (GroupSecurityContextConstraintsStrategy, error) {
+func NewMustRunAs(ranges []securityv1.IDRange, field string, groupAccessor GroupAccessorFunc, groupMutator GroupMutatorFunc) (GroupSecurityContextConstraintsStrategy, error) {
+	if groupAccessor == nil {
+		return nil, fmt.Errorf("function describing accessing groups is required")
+	}
+
 	if len(ranges) == 0 {
 		return nil, fmt.Errorf("ranges must be supplied for MustRunAs")
 	}
 	return &mustRunAs{
-		ranges: ranges,
-		field:  field,
+		ranges:        ranges,
+		field:         field,
+		groupAccessor: groupAccessor,
+		groupMutator:  groupMutator,
 	}, nil
 }
 
 // Generate creates the group based on policy rules.  By default this returns the first group of the
 // first range (min val).
-func (s *mustRunAs) Generate(_ *api.Pod) ([]int64, error) {
-	return []int64{s.ranges[0].Min}, nil
+func (s *mustRunAs) MutatePod(podSC securitycontext.PodSecurityContextMutator) error {
+	if s.groupMutator == nil {
```

Comment:
i don't see a call to `group.NewMustRunAs` where `groupMutator` is actually `nil`.